Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/CoronaTestQRCode.kt

```diff
@@ -11,31 +12,46 @@ import org.joda.time.LocalDate
 sealed class CoronaTestQRCode : Parcelable, TestRegistrationRequest {
 
     abstract override val type: CoronaTest.Type
-    abstract val qrCodeGUID: CoronaTestGUID
-
-    @IgnoredOnParcel
-    override val identifier: String
-        get() = "qrcode-${type.raw}-$qrCodeGUID"
+    abstract val registrationIdentifier: String
 
     @Parcelize
     data class PCR(
-        override val qrCodeGUID: CoronaTestGUID,
+        val qrCodeGUID: CoronaTestGUID,
     ) : CoronaTestQRCode() {
 
-        @IgnoredOnParcel override val type: CoronaTest.Type = CoronaTest.Type.PCR
+        @IgnoredOnParcel
+        override val type: CoronaTest.Type = CoronaTest.Type.PCR
+
+        @IgnoredOnParcel
+        override val identifier: String
+            get() = "qrcode-${type.raw}-$qrCodeGUID"
+
+        @IgnoredOnParcel
+        override val registrationIdentifier: String
+            get() = qrCodeGUID
     }
 
     @Parcelize
     data class RapidAntigen(
-        override val qrCodeGUID: CoronaTestGUID,
+        val hash: RapidAntigenHash,
         val createdAt: Instant,
         val firstName: String?,
         val lastName: String?,
         val dateOfBirth: LocalDate?,
     ) : CoronaTestQRCode() {
 
-        @IgnoredOnParcel override val type: CoronaTest.Type = CoronaTest.Type.RAPID_ANTIGEN
+        @IgnoredOnParcel
+        override val type: CoronaTest.Type = CoronaTest.Type.RAPID_ANTIGEN
+
+        @IgnoredOnParcel
+        override val identifier: String
+            get() = "hash-${type.raw}-$hash"
```

Comment:
Could we call the prefix here also `qrcode` as with PCR? Both are sourced from a QRCode.